meta.lua

```diff
@@ -638,37 +638,109 @@ local Player = {
     get_lighting = function(self) end,
 }
 
+---@alias ObjectVisuals
```

Comment:
I added a bunch of stuff to meta.lua that might relate to Entities, because it was also tangentially related to PenTool.

PenTool is not an entity.

Since meta.lua is not actually run, and is just autocomplete hints, I feel fine about mingling PR jobs a bit, particularly because we don't actually have a issue for improving meta.lua It's just being improved as we find holes.